zeppelin-web/src/app/notebook/paragraph/paragraph.controller.js

```diff
@@ -865,6 +865,29 @@ angular.module('zeppelinWebApp').controller('ParagraphCtrl', function($scope, $r
     }
   };
 
+  $scope.parseTableCell = function(cell) {
+    //here we try to automatically parse the data type for each
+    //cell in the table, i.e. string/number/date
+    // 1. is this a number (possibly)?
+    if (!isNaN(cell)) {
+      return cell.length === 0 ? cell : ((Number(value) === Number.NaN || Number(value) > Number.MAX_SAFE_INTEGER || Number(value) < Number.MIN_SAFE_INTEGER) ? cell : Number(cell));
+    }
+    // 2. is this a date?
```

Comment:
Actually there is a lot of cases that can go wrong,
06/08/2016 could be both DD/MM/YYYY or MM/DD/YYYY
Maybe letting the user specify the Date format could be added as part of #1363 

Btw @Peilin-Yang moment.js actually accept an array of formats, so you could even remove the loop
(cf. https://stackoverflow.com/questions/7445328/check-if-a-string-is-a-date-value/30870755#30870755)